apps/contrib/formset.py

```diff
@@ -0,0 +1,22 @@
+from django import forms
+
+
+class DynamicModelFormSet(forms.BaseModelFormSet):
+    def get_default_prefix(cls):
+        return 'test'
+
+    @property
+    def media(self):
+        additional_js = forms.Media(js={
+            'formset.js'
+        })
+        return super().media + additional_js
+
+
+def dynamic_modelformset_factory(*args, **kwargs):
+    kwargs.update({
+        'formset': DynamicModelFormSet
+    })
+    modelformset = forms.modelformset_factory(*args, **kwargs)
+
+    return modelformset
```

Comment:
You could just add the js file inside the template, but the thought behind this was to make it more reusable and especially easier to use, because it takes care care of implementing the JS.